source/CaptureClass.cpp

```diff
@@ -0,0 +1,107 @@
+/* CaptureClass.cpp
+Copyright (c) 2023 by Amazinite
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "CaptureClass.h"
+
+#include "DataNode.h"
+#include "DataWriter.h"
+
+#include <algorithm>
+
+using namespace std;
+
+
+CaptureClass::CaptureClass(const DataNode &node)
+{
+	name = node.Token(1);
+
+	for(const DataNode &child : node)
+	{
+		if(child.Size() < 2)
+			child.PrintTrace("Skipping invalid capture class key with no value:");
+		const string &key = child.Token(0);
+		double value = child.Value(1);
+		if(key == "success")
+			successChance = max(0., min(1., value));
```

Comment:
So you can have a tool with 0% succes chance?
I assume that all attributes will be displayed, right? So it seems a bit weird because the boarding will happen as if you could do smth but you wont?
Same thing for the ship having 0% cap chance itself, it's kind of weirdly uncapable in that case.
Do we just accept mis-use or should we put safegards/warnings?

Reply:
C++17 introduce [std::clamp](https://en.cppreference.com/w/cpp/algorithm/clamp)